### PR TITLE
Rework errors to store code, message, severity, and variable parts

### PR DESCRIPTION
**Pull request by Envek**
To support translation of error messages

Error params stores names of specifics, amounts of money, etc.

I've bumped version as this is breaking change. The way of how errors stored inside exceptions is changed.

### CODE REVIEW

File: lib/ebay_request/error_item.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# A single error from many returned in API response
+class EbayRequest::ErrorItem
```
Comment (nepalez) [inline lib/ebay_request/error_item.rb:4]: Why not use dry-initializer interface here, which is already in the gem's dependencies?

```ruby
class EbayRequest::ErrorItem
  extend Dry::Initializer

  option :errorId,  proc(&:to_s), as: :code
  option :message,  proc(&:to_s)
  option :severity, proc(&:to_s)
  option :params,   proc(&:to_h), default: -> {}

  alias :to_s :message

  def inspect
    "#<#{self.class.name} #{severity} #{code}: #{message.inspect}>"
  end

  def self.new(source)
    # accept source with string keys
    source = source.to_h.each_with_object({}) { |(k, v), obj| obj[k.to_sym] = v }
    super(source)
  end
end
```

File: lib/ebay_request/business_policies.rb [hunk 2]
```diff
@@ -49,8 +49,12 @@ def auth_header
   end
 
   def errors_for(response)
-    [response.dig("errorMessage", "error")].flatten.compact.map do |err|
-      [err["severity"], err["errorId"], err["message"]]
+    [response.dig("errorMessage", "error")].flatten.compact.map do |error|
+      EbayRequest::ErrorItem.new(
```
Comment (nepalez) [inline lib/ebay_request/business_policies.rb:53]: With dry-initializer-based item you can use just:

```ruby
EbayRequest::ErrorItem.new(error)
```

File: lib/ebay_request/finding.rb [hunk 3]
```diff
@@ -27,10 +27,13 @@ def headers(callname)
   end
 
   def errors_for(response)
-    [response.dig("errorMessage", "error")]
-      .flatten
-      .compact
-      .map { |err| [err["severity"], err["errorId"], err["message"]] }
+    [response.dig("errorMessage", "error")].flatten.compact.map do |error|
+      EbayRequest::ErrorItem.new(
```
Comment (nepalez) [inline lib/ebay_request/finding.rb:31]: With dry-initializer-based item you can use just:

```ruby
EbayRequest::ErrorItem.new(error)
```

File: lib/ebay_request/error_item.rb (hunk 1 above)
Comment (Envek) [reply lib/ebay_request/error_item.rb:4]: Oh yes, didn't notice that it is available. Changed to use it.

Decided not to use `errorId` as only half of APIs use this name. Other half uses `errorCode`. I think it is better not to bound naming to any single API

File: lib/ebay_request/finding.rb (hunk 3 above)
Comment (Envek) [reply lib/ebay_request/finding.rb:31]: Different APIs have different error format. It is better to keep explicit mapping IMO.

File: lib/ebay_request/business_policies.rb (hunk 2 above)
Comment (Envek) [reply lib/ebay_request/business_policies.rb:53]: Different APIs have different error format. It is better to keep explicit mapping IMO.

**Pull request merged**